core/state_processor.go

```diff
@@ -67,7 +67,12 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 	}
 
 	if random.IsRunning() {
-		err := random.RevealAndCommit(block.Randomness().Revealed, block.Randomness().Committed, header.Coinbase, header, statedb)
+		author, err := p.bc.Engine().Author(header)
```

Comment:
just to point out here, that in the old code (which some nodes will be running) if the block's Coinbase != Author it will fail.

On the new version it doesn't matter since we use header author which makes the change retrocompatible. But we need to be very carefully that we don't generate blocks BEFORE the hardfork where Coinbase != Author. Since they are assumed to be the same PREFORK.